modules/dag-l1/src/main/scala/org/tessellation/dag/l1/domain/snapshot/programs/SnapshotProcessor.scala

```diff
@@ -281,21 +283,19 @@ abstract class SnapshotProcessor[
 
                           if (!areTipsAligned)
                             MonadThrow[F].raiseError[Alignment](TipsGotMisaligned(deprecatedTipsToAdd, tipsToDeprecate))
-                          else if (onlyInMajority.isEmpty)
-                            Applicative[F].pure[Alignment](
-                              AlignedAtNewOrdinal(
-                                snapshot,
-                                state,
-                                toMarkMajority.toSet,
-                                tipsToDeprecate,
-                                tipsToRemove,
-                                txRefsToMarkMajority,
-                                postponedToWaiting
-                              )
-                            )
                           else
-                            Applicative[F]
-                              .pure[Alignment](
+                            isAlignedWithGlobalL0.map { aligned =>
```

Comment:
You will redownload to a snapshot that is being processed, so it doesn't give you anything in terms of chain repairing. The snapshots that get here need to form a correct chain and if you detected externally that that chain locally is already wrong why redownload to the snapshot from that wrong chain.

Essentially I think that the snapshots that we process should be picked during pull from GL0. There is a `L0Service` that handles downloading snapshot from Global L0 I think you should improve (or extend) it to allow injecting a download strategy, so that the snapshots that are not perceived as majority don't get downloaded in the first place (hence no fork will happen and there will be no need for fixing it). You could also add a filter function on the pulled snapshots stream between L0Service and SnapshotProcessor filtering out snapshots that are not majority, but I think that we shouldn't download snapshots that are not perceived (by your local logic) as majority in the first place. The processor should get a already correct (majority) snapshot and only handle adjusting local state if it accepted some blocks that did not get accepted by GL0.

Regarding the `isAlignedWithGlobalL0` function I think only stating true or false is too little, the process here doesn't know when it forked and what is the majority hash to redownload to. Anyway I think this part should not be needed as processes before the `SnapshotProcessor` should already pick the majority correctly and it should not be `SnapshotsProcessor`'s responsibility.

Reply:
@TheMMaciek Is your idea that we perform that filter in `GlobalL0Service.pullGlobalSnapshots` ?
That seems to be the call that hands snapshots to the processor.

cc: @ryle-goehausen

Reply:
Has this comment already been addressed sufficiently?

Reply:
Resolving; code no longer present